core/src/main/java/org/radargun/stages/test/RunningTest.java

```
@@ -0,0 +1,290 @@
+         return null;
+      } else {
+         statisticsCounter.incrementAndGet();
+         return statisticsPrototype.newInstance();
+      }
+   }
+
+   public void recordStatistics(Statistics stats) {
+      statisticsQueue.add(stats);
+   }
+
+   public List<Statistics> getStatistics() {
+      interruptStressors();
+      ArrayList<Statistics> statistics = new ArrayList<>();
+      while (!terminated && statistics.size() < statisticsCounter.get()) {
+         Statistics stats = statisticsQueue.poll();
+         if (stats != null) {
+            statistics.add(stats);
+         } else {
+            try {
+               Thread.sleep(1000);
+            } catch (InterruptedException e) {
+               throw new IllegalStateException(e);
+            }
+         }
+      }
+      statisticsCounter.set(0);
+      return statistics;
+   }
+
+   protected void interruptStressors() {
+      synchronized (this) {
+         for (Stressor stressor : stressors) {
+            stressor.interrupt();
+         }
+      }
+   }
+
+   public void stopStressors() {
+      finished = true;
+      steadyState = false;
+      interruptStressors();
+      List<Stressor> stressors;
+      synchronized (this) {
+         stressors = new ArrayList<>(this.stressors);
+      }
+      for (Stressor stressor : stressors) {
+         try {
+            stressor.join();
+         } catch (InterruptedException e) {
+            throw new IllegalStateException(e);
+         }
+      }
+      synchronized (this) {
+         for (Runnable listener : stopListeners) {
+            listener.run();
+         }
+      }
+   }
+
+   public void addStressor(boolean failSilently) {
+      int threadIndex = getNextThreadId(maxThreads);
+      if (threadIndex < 0) {
+         if (failSilently) {
+            return;
+         }
+         reachedMax = true;
+         log.warnf("Attempt to create more than %d threads!", maxThreads);
+         Utils.threadDump();
+         return;
```

Shouldn't this throw an exception instead? Otherwise this fails silently even if failSilently == false.